src/main/java/com/google/sps/data/LearningPathService.java

```diff
@@ -183,35 +168,27 @@ private ItemFeedback mapEntityToItemFeedback(long sectionId, Entity e) {
 	}
 
 
-    public long getSectionCompletion(long userId, long sectionId ,long learningPath)throws EntityNotFoundException { 
+    public double getSectionCompletion(long userId, LearningSection section ,long learningPath)throws EntityNotFoundException { 
        // filter 
         List<ItemFeedback> feedbacks = loadItemFeedbacksUser(userId).stream()
-                                    .filter( f -> f.isCompleted()).collect(Collectors.toList());
-        List<LearningItem> itemsInSection = loadItems(sectionId);
-
-        List<Long> feedbackId = feedbacks.stream().map(p -> p.getLearningItem())
-                                .collect(Collectors.toList()); 
-
-        List<Long> sectionIds = itemsInSection.stream().map(p -> p.getId())
-                                .collect(Collectors.toList()); 
-       
-        feedbackId.retainAll(sectionIds);
+                                    .filter( f -> f.getLearningSection() == section.getId() && f.isCompleted()).collect(Collectors.toList());
```

Comment:
In addition to `.collect(..)`, Java streams also support a `.count()` method. This is more direct in cases where the actual items don't matter:

https://docs.oracle.com/javase/8/docs/api/java/util/stream/Stream.html#count--